lib/rspec/support/object_inspector.rb

```diff
@@ -13,6 +13,8 @@ def self.inspect(object)
           "#{object.to_s 'F'} (#{object.inspect})"
         elsif RSpec::Support.is_a_matcher?(object) && object.respond_to?(:description)
           object.description
+        elsif object.class == RSpec::Mocks::Double
+          "Double (#{object.instance_variable_get(:@name)})"
```

Comment:
> @JonRowe I think we need to add `attr_reader :name` on `RSpec::Mocks::Double` to avoid prying into the object. Thoughts please?

We intentionally have not exposed a `name` attribute off of `RSpec::Mocks::Double` because doubles are meant to be blank slate objects that have as few methods as possible initially defined so that the user is given free reign to define whatever interface they want w/o us encroaching on the interface w/ our own methods.  The user needs to be able to stub `name` for their own use.

So I think we need to pry into the double to get the instance variable, but doing so _from a completely separate library_ (rspec-support) feels wrong, and that's where my suggestion above helps: by providing an interface from rspec-support where rspec-mocks can register doubles with `ObjectInspector`, we can keep this logic in rspec-mocks where it's not nearly as big of a deal.